src/components/login/login.agency.test.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import Login from './login.component';
+import { render, screen, waitFor, userEvent } from '../../testUtils'
+
+import { CHARTS_PAGE } from "../../root/root.constants";
+
+import i18next from "../../helpers/i18n/index";
+
+const MOCK_CORRECT_USERNAME = "user@example.com"
+const MOCK_CORRECT_PASSWORD = "password"
+
+const loginLabel = i18next.t("LOGIN_PAGE.EMAIL_USERNAME")
+const passwordLabel = i18next.t("LOGIN_PAGE.PASSWORD")
+const loginButton = i18next.t("LOGIN_PAGE.LOGIN")
+
+// --- mock AuthService ---
+const mockLoginSuccess = jest.fn()
+
+jest.mock('../../services/auth.service', () => ({
+    getInstance: jest.fn().mockImplementationOnce(() => ({
+        // FIXME: find a way to mock the userRole dynamically between tests in the same file
```

Comment:
I can't find a way to mock ES6 class' static function and change its mock inside a file, that's why this part is duplicated across 3 files: one for staff, one for global admin, one for agency admin